lib/ansible/module_utils/facts.py

```diff
@@ -1621,6 +1623,58 @@ def get_dmi_facts(self):
             if found:
                 self.facts['product_name'] = found.group(1)
 
+    def get_device_facts(self):
+        self.facts['devices'] = {}
+
+        disk_stats = {
+            'Product': 'product',
+            'Revision': 'revision',
+            'Serial No': 'serial',
+            'Size': 'size',
+            'Vendor': 'vendor',
+            'Hard Errors': 'hard_errors',
+            'Soft Errors': 'soft_errors',
+            'Transport Errors': 'transport_errors',
+            'Media Error': 'media_errors',
+            'Predictive Failure Analysis': 'predictive_failure_analysis',
+            'Illegal Request': 'illegal_request',
+        }
+
+        cmd = ['/usr/bin/kstat', '-p']
+
+        for ds in disk_stats:
+            cmd.append('sderr:::%s' % ds)
+
+        d = {}
+        rc, out, err = self.module.run_command(cmd)
+        if rc != 0:
+            return dict()
+
+        sd_instances = set(sorted([line.split(':')[1] for line in out.split('\n') if line.startswith('sderr')]))
```

Comment:
* You don't need sorted here because a set has no order.
* You can use a generator expression instead of a list comprehension
* I encourage the use of frozenset when the data is not going to be modified in-place
``` python
sd_instances = frozenset(line.split(':')[1] for line in out.split('\n') if line.startswith('sderr'))
```